libBlueSky/Dialog.cpp

```diff
@@ -131,13 +133,92 @@ namespace BlueSky {
         delete w;
     }
 
+    bool Dialog::isResizerEnabled() const
+    {
+        return mSizeGripEnabled;
+    }
+
+    void Dialog::setResizerEnabled(bool enabled)
+    {
+        mSizeGripEnabled = enabled;
+    }
+
+    void Dialog::open()
+    {
+#ifndef Q_OS_MAC
+        setWindowFlags(windowFlags() | Qt::FramelessWindowHint | Qt::Sheet);
+        new SheetEventFilter(parentWidget(), this);
+#endif
+
+        QDialog::open();
+    }
+
+#ifndef Q_OS_MAC
+    void Dialog::moveEvent(QMoveEvent* ev)
+    {
+        QDialog::moveEvent(ev);
+        if (isSheet()) {
+            updatePinnedPos();
+        }
+    }
+
+    void Dialog::resizeEvent(QResizeEvent* ev)
+    {
+        if (mResizer) {
+            mResizer->updatePos();
+            mResizer->raise();
+        }
+
+        QDialog::resizeEvent(ev);
+    }
+#endif
+
     void Dialog::showEvent(QShowEvent* ev) {
         w->applyStyle(this);
         QDialog::showEvent(ev);
+
+        if (mSizeGripEnabled) {
+            if (!mResizer) {
+                mResizer = new SizeGrip(this);
+            }
+#ifndef Q_OS_MAC
+            if (isSheet()) {
+                mResizer->setResizeBehaviour(SizeGrip::ResizeMirrorHorizontal);
+            }
+#endif
+            mResizer->updatePos();
+            mResizer->raise();
+            mResizer->show();
+        }
     }
 
     void Dialog::watchDogBark() {
         w->bark();
     }
 
+#ifndef Q_OS_MAC
+    SheetEventFilter::SheetEventFilter(QWidget* topLevelWidget, Dialog* parent)
+        : QObject(parent)
+        , mParentDlg(parent)
+    {
+        Q_ASSERT(parent);
+        Q_ASSERT(topLevelWidget && topLevelWidget->isWindow());
+        topLevelWidget->installEventFilter(this);
+    }
+
+    bool SheetEventFilter::eventFilter(QObject* obj, QEvent* event)
+    {
+        switch (event->type()) {
+        default: break;
+
+        case QEvent::Move:
+        case QEvent::Resize:
+            mParentDlg->updatePinnedPos();
+            return true;
```

Comment:
Hm ... like expected, `QObject::eventFilter` returns false. Anyway the `break` seems to be more correct.